backend/src/ml_space_lambda/app_configuration/lambda_functions.py

```diff
@@ -18,39 +18,234 @@
 import logging
 import time
 
+import boto3
+
 from ml_space_lambda.data_access_objects.app_configuration import AppConfigurationDAO, AppConfigurationModel, SettingsModel
-from ml_space_lambda.utils.common_functions import api_wrapper
+from ml_space_lambda.data_access_objects.resource_metadata import ResourceMetadataDAO
+from ml_space_lambda.enums import EnvVariable, IAMEffect, IAMStatementProperty, ResourceType, ServiceType
+from ml_space_lambda.utils.account_utils import get_account_arn
+from ml_space_lambda.utils.common_functions import (
+    api_wrapper,
+    event_wrapper,
+    generate_exception_response,
+    generate_html_response,
+    retry_config,
+)
+from ml_space_lambda.utils.iam_manager import IAM_RESOURCE_PREFIX, IAMManager
+from ml_space_lambda.utils.mlspace_config import get_environment_variables
+from ml_space_lambda.utils.resoruce_utils import suspend_all_of_type
 
+iam_manager = IAMManager()
 log = logging.getLogger(__name__)
 app_configuration_dao = AppConfigurationDAO()
+env_variables = get_environment_variables()
+translate_client = boto3.client("translate", config=retry_config)
+resource_metadata_dao = ResourceMetadataDAO()
 
+service_group_disable_permissions = [
+    {
+        "Description": "Permissions shared by translate services",
+        "ServiceGroup": [ServiceType.BATCH_TRANSLATE, ServiceType.REALTIME_TRANSLATE],
+        "Statements": [
+            {
+                IAMStatementProperty.EFFECT: IAMEffect.DENY,
+                IAMStatementProperty.ACTION: [
+                    "translate:ListTerminologies",
+                    "translate:ListLanguages",
+                    "comprehend:Detect*",
+                    "comprehend:BatchDetect*",
+                ],
+                IAMStatementProperty.RESOURCE: "*",
+            },
+            {
+                IAMStatementProperty.EFFECT: IAMEffect.DENY,
+                IAMStatementProperty.ACTION: "iam:PassRole",
+                IAMStatementProperty.RESOURCE: get_account_arn("iam", f"role/{IAM_RESOURCE_PREFIX}*"),
+                IAMStatementProperty.CONDITION: {"StringEquals": {"iam:PassedToReception": "translate.amazonaws.com"}},
+            },
+        ],
+    }
+]
 
-@api_wrapper
-def update_configuration(event, context):
-    request = json.loads(event["body"])
-    configScope = request["configScope"]
-    version_id = request["versionId"] + 1  # increment so this will be the latest version
+service_disable_permissions = {
+    ServiceType.BATCH_TRANSLATE: {
+        "Statements": [
+            {
+                IAMStatementProperty.EFFECT: IAMEffect.DENY,
+                IAMStatementProperty.ACTION: [
+                    "translate:StopTextTranslationJob",
```

Comment:
This applies to everything, but why can't we just say

```
translate:*
```

For all of these things? It seems like there are some actions that we may need to be more fine-grained with, but with something like translate or EMR can we just wildcard everything?